quic/ngtcp2/errors.nim

```diff
@@ -3,5 +3,5 @@ import ngtcp2
 type Ngtcp2Error* = object of IOError
 
 proc checkResult*(result: cint) =
-  if result != 0:
+  if result < 0:
     raise newException(Ngtcp2Error, $ngtcp2_strerror(result))
```

Comment:
I've never seen this style of constructing a ref before, and I had to think twice to understand what's happening. So I'm inclined to leave the newException in place, because the name makes it obvious that it's creating a new Exception.

Reply:
this is part of the reason to avoid `ref object` in general, and instead use `ref` explicitly on the consuming side - ie when to type `Connection(x: .., y: ...)` right now, you have no clue that this will create a ref object - same with `func f(v: Connection)` - there's no hint there that connection is a ref.. - it all becomes clear when you stop using `ref object`, and instead pass `func f(v: ref Connection)`  where ref semantics are desired, and construct with `(ref Connection)` - `new` has the problem that it _forces_ you to initialize the object field by field which quickly gets messy (when exceptions happen in the middle for example)

`ref object` this that kind of shortcut that soon becomes a detour

Reply:
Yes, I agree that it is better to avoid ref object in many cases. Yet here, with `Connection`, I purposely do not want the user of the library to be able to work with non-ref Connections. Connection holds a pointer to the ngtcp2_conn struct. I free that struct in  `destroy` after checking that the pointer is not already set to nil. If the user were to use the non-ref object, they would create multiple copies of the object and the pointer it contains, which upon destruction would lead to a double free.